mempool/clist_mempool.go

```diff
@@ -130,6 +149,16 @@ func (mem *CListMempool) EnableTxsAvailable() {
 	mem.txsAvailable = make(chan struct{}, 1)
 }
 
+func (mem *CListMempool) SetTxRemovedCallback(cb func(txKey types.TxKey)) {
+	mem.removeTxOnReactor = cb
+}
+
+func (mem *CListMempool) invokeRemoveTxOnReactor(txKey types.TxKey) {
+	if mem.removeTxOnReactor != nil {
```

Comment:
I would report an error if the callback is null here. This should never happen.

Reply:
I added a comment.